src/Microsoft.Identity.Client/Utils/EnumerableExtensions.cs

```diff
@@ -30,15 +31,15 @@ internal static bool ContainsOrdinalIgnoreCase(this IEnumerable<string> set, str
             return set.Any(el => el.Equals(toLookFor, System.StringComparison.OrdinalIgnoreCase));
         }
 
-        internal static List<T> FilterWithLogging<T>(
-            this List<T> list,
+        internal static IEnumerable<T> FilterWithLogging<T>(
+            this IEnumerable<T> list,
             Func<T, bool> predicate,
             ICoreLogger logger,
             string logPrefix)
         {
-            int numberBefore = list.Count;
+            int numberBefore = list.Count();
```

Comment:
does IEnumerable even support Count()?  This would walk through the entire IEnumerable to do this and then you'll enumerate it again in the Where clause.  That's why this was List<T> previously.

Reply:
It won't. The implementation of the Count() extension property will use the underlying collection's Count property if it is available. Otherwise the collection will be enumerated to calculate the count.